tchap_red_list/__init__.py

```diff
@@ -79,16 +88,30 @@ async def update_red_list_status(
         # Compare what status (in the list, not in the list) the user wants to have with
         # what it already has. If they're the same, don't do anything more.
         desired_status = bool(content.get("hide_profile"))
-        current_status, _ = await self._get_user_status(user_id)
+        current_status, because_expired = await self._get_user_status(user_id)
 
-        if current_status == desired_status:
+        # If the user's desired status is the same as their current one, don't do
+        # anything, except if because_expired is True (because in this case it can be a
+        # user that was added to the red list after expiring, haven't been removed yet,
+        # and manually re-added themselves).
+        if current_status == desired_status and because_expired is False:
             return
 
-        # Add or remove the user depending on whether they want their profile hidden.
-        if desired_status is True:
-            await self._add_to_red_list(user_id)
+        if current_status == desired_status:
+            if because_expired is True:
```

Comment:
Did you mean to leave the first `if` in?

Reply:
Ah I didn't, thanks.